iseq.c

```diff
@@ -974,13 +979,8 @@ rb_iseq_new_with_opt(const rb_ast_body_t *ast, VALUE name, VALUE path, VALUE rea
         option = set_compile_option_from_ast(&new_opt, ast);
     }
 
-    VALUE script_lines = Qnil;
-
-    if (ast && !FIXNUM_P(ast->script_lines) && ast->script_lines) {
-        script_lines = ast->script_lines;
-    }
-    else if (parent) {
-        script_lines = ISEQ_BODY(parent)->variable.script_lines;
+    if (NIL_P(script_lines) && ast && ast->script_lines && !FIXNUM_P((VALUE)ast->script_lines)) {
```

Comment:
Do we need to check `!FIXNUM_P(script_lines)` too?
By the way, is it possible to make the call of this function always pass meaningful value for `VALUE script_lines`? If so, no need to check

Reply:
I reverted the possible over-corrected code